main.go

```diff
@@ -293,9 +308,53 @@ func checkParams(c *cli.Context) error {
 		return fmt.Errorf("Missing required param: cluster")
 	}
 
+	if !isValidKubectlVersionParam(c, extraKubectlVersions) {
+		return fmt.Errorf(getInvalidKubectlVersionMessage(c, extraKubectlVersions))
+	}
+
 	return nil
 }
 
+// isValidKubectlVersionParam tests whether a given version is valid within the current environment
+func isValidKubectlVersionParam(c *cli.Context, availableVersions []string) bool {
```

Comment:
The way I would structure this function is to return `error`, then simply check the `err` in the calling location to see if it is nil or not:
```go
if err := validKubectlVersion(c.String("kubectl-version"), extraKubectlVersions); err != nil {
    return err
}
```

This collapses the checking and the error message cases into one call, vs traversing the logic in both places.

Reply:
awesome feedback. thanks!